src/main/cpp/src/api/genomicsdb.cc

```diff
@@ -725,3 +802,665 @@ void GenomicsDBVariantCallProcessor::process(const std::string& sample_name,
+  int16_t one = 1;
+  bgen_file.write((char*)&one, 2); // BGEN: 2 byte length of variant identifier, not stored in GenomicsDB so using dummy
+  bgen_file.write((char*)&one, 1); // BGEN: dummy variant id
+  int16_t rsid_len = rsid.length();
+  bgen_file.write((char*)&rsid_len, 2); // BGEN: 2 byte length of rsid
+  bgen_file.write(rsid.c_str(), rsid_len); // BGEN: rsid
+  std::string chrom = genomic_interval.contig_name;
+  int16_t chrom_len = chrom.length();
+  bgen_file.write((char*)&chrom_len, 2); // BGEN: 2 byte chrom length
+  bgen_file.write(chrom.c_str(), chrom_len); // BGEN: chrom
+  uint32_t variant_pos = genomic_interval.interval.first;
+  bgen_file.write((char*)&variant_pos, 4); // BGEN: 4 byte variant position
+  int16_t K = vec.size();
+  bgen_file.write((char*)&K, 2);// BGEN: 2 byte K for number of alleles
+  // write alleles and lengths
+  int32_t len;
+  for(auto& a : vec) { // iterate over alleles
+    len = a.length();
+    bgen_file.write((char*)&len, 4); // BGEN: 4 byte length of allele
+    bgen_file.write(a.c_str(), len); // BGEN: allele
+  }
+
+  // BGEN: genotype data block (layout 2, uncompressed)
+  bgen_gt_size_offset = bgen_file.tellp();
+  bgen_gt_size = 0;
+  int32_t fourB_zero = 0;
+
+  // BGEN: preallocate probability data storage
+  int32_t N = sample_map.size();
+  codec_buf.append((char*)&N, 4); // BGEN: 4 byte N
+  codec_buf.append((char*)&K, 2); // BGEN: 2 byte K
+  bgen_min_ploidy_offset = bgen_file.tellp();
+  codec_buf.append((char*)&fourB_zero, 1); // BGEN: 1 byte min ploidy (placeholder)
+  bgen_max_ploidy_offset = bgen_file.tellp();
+  codec_buf.append((char*)&fourB_zero, 1); // BGEN: 1 byte max ploidy (placeholder)
+  bgen_ploidy_info_offset = bgen_file.tellp();
+  char default_sample_info = 0b10000010; // default missingness and ploidy information: set to missing/diploid unspecified
+  for(int j = 0; j < N; j++) {
+    codec_buf.append(&default_sample_info, 1); // BGEN: default sample information within this variant: because missing is set to 1, no need to backfill skipped cells
+  }
+  codec_buf.append((char*)&phased, 1); // BGEN: 1 byte phased
+  char B = 8; // precision at one byte to avoid alignment difficulties
+  codec_buf.append(&B, 1); // BGEN: 1 byte unsigned bits of precision
+  bgen_probability_offset = bgen_file.tellp();
+}
+
+void GenomicsDBPlinkProcessor::process(const Variant& variant) {
+  auto& calls = variant.get_calls();
+
+  bool phased = true;
+  std::string gt_string;
+
+  for(auto& c : calls) {
+    auto fields = get_genomic_fields_for(array, &c, query_config);
+    
+    for(auto& f : fields) {
+      if(f.name == "GT") {
+        gt_string = f.to_string(get_genomic_field_type(f.name));
+
+        if(gt_string.find('/') != std::string::npos && gt_string.find('.') == std::string::npos) { // unphased if contains a slash or is ./. (might be able to use GL/PL, which are unphased probabilities)
+          phased = false;
+        }
+        break;
+      }
+    }
+  }
```

Comment:
So this loop goes through all the calls and looks to see if any of them are unphased? Wouldn't it be best to stop iterating on calls once we see the first unphased?